### ISSUE TITLE
dotnet publish throws NRE

### ISSUE THREAD

**Issue by moozzyk** [dotnet/cli#2480]
## Steps to reproduce
```
PS C:\temp\reprobreaking> dotnet.exe -v publish -o C:\temp\reprobreaking1 --framework netstandardapp1.5
Object reference not set to an instance of an object.
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.DotNet.ProjectModel.ProjectContextBuilder.Build()
   at Microsoft.DotNet.Tools.Publish.PublishCommand.SelectContexts(String projectPath, NuGetFramework framework, String runtime)
   at Microsoft.DotNet.Tools.Publish.PublishCommand.TryPrepareForPublish()
   at Microsoft.DotNet.Tools.Publish.PublishCommand.<>c__DisplayClass0_0.<Run>b__0()
   at Microsoft.Dnx.Runtime.Common.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.DotNet.Tools.Publish.PublishCommand.Run(String[] args)
```

## Expected  behavior
publish works

## Actual behavior

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.DotNet.ProjectModel.ProjectContextBuilder.Build()
   at Microsoft.DotNet.Tools.Publish.PublishCommand.SelectContexts(String projectPath, NuGetFramework framework, String runtime)
   at Microsoft.DotNet.Tools.Publish.PublishCommand.TryPrepareForPublish()
   at Microsoft.DotNet.Tools.Publish.PublishCommand.<>c__DisplayClass0_0.<Run>b__0()
   at Microsoft.Dnx.Runtime.Common.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.DotNet.Tools.Publish.PublishCommand.Run(String[] args)
```
## Environment data
`dotnet --info` output:

PS C:\temp\reprobreaking> dotnet --version
1.0.0-rc2-002376

**Comment by sokket** [comment]
Can you post a ```dir``` of what is in your test directory?

**Comment by moozzyk** [comment]
It did not have project.json (I think it was empty) - I just forgot to go back to the project directory from my first publish output.

**Comment by sokket** [comment]
Got it, so ```C:\temp\reprobreaking``` was empty and you ran the above command, causing this error (just to clarify)?

**Comment by moozzyk** [comment]
yes

**Comment by sokket** [comment]
Cool, thanks for clarifying!

**Comment by sokket** [comment]
Fixed via #2538
